gossip/client/namedtree.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"strings"
+
+	"github.com/quorumcontrol/chaintree/chaintree"
+	"github.com/quorumcontrol/chaintree/nodestore"
+	"github.com/quorumcontrol/messages/v2/build/go/transactions"
+	"github.com/quorumcontrol/tupelo-go-sdk/consensus"
+)
+
+type Generator struct {
+	Namespace string
+	Client    *Client
+}
+
+type NamedChainTree struct {
+	Name      string
+	ChainTree *consensus.SignedChainTree
+	Client    *Client
+
+	genesisKey *ecdsa.PrivateKey
+	nodeStore  nodestore.DagStore
+	owners     []string
+}
+
+type NamedChainTreeOptions struct {
+	Name              string
+	ObjectStorageType string
+	Client            *Client
+	NodeStore         nodestore.DagStore
```

Comment:
it would be weird if this was different than the client's?

Reply:
client didn’t expose the DagStore until recently and was after @cap10morgan had written this code in dgit - agreed though, it can just pull from client now